plugins/generator-1.19.4/forge-1.19.4/templates/livingentity/livingentity_renderer.java.ftl

```diff
@@ -109,9 +109,21 @@ public class ${name}Renderer extends <#if humanoid>Humanoid</#if>MobRenderer<${n
 		</#if>
 	}
 
-	<#if data.mobModelName == "Villager">
-	@Override protected void scale(${name}Entity villager, PoseStack poseStack, float f) {
-		poseStack.scale(0.9375f, 0.9375f, 0.9375f);
+	<#if data.mobModelName == "Villager" || (data.visualScale?? && (data.visualScale.getFixedValue() != 1 || hasProcedure(data.visualScale)))>
+	@Override protected void scale(${name}Entity entity, PoseStack poseStack, float f) {
+		<#if hasProcedure(data.visualScale)>
+			Level world = entity.level;
+			double x = entity.getX();
+			double y = entity.getY();
+			double z = entity.getZ();
+			float scale = (float) <@procedureOBJToNumberCode data.visualScale/>;
+			poseStack.scale(scale, scale, scale);
+		<#else>

```

Comment:
This generates excessive code when used in combination with villager model, the condition should be corrected a bit:
```suggestion
		<#elseif data.visualScale?? && (data.visualScale.getFixedValue() != 1>
```
Same for the other version.

Reply:
Isn't this handled by the parent if?

Reply:
The outer `if` has a condition for whether there is fixed scale value, dynamic scale value or model set to `Villager`. The `if` before this `else` only checks for dynamic scale so this branch is included even if model is matching but fixed scale isn't.

Reply:
Good catch, this is why I value bug-finding reviews so much! :) This would be a patch update very likely if you wouldn't catch it :D

Would you mind offering suggestion to the other file too and I can just  ![image](https://github.com/MCreator/MCreator/assets/16374228/41c6df47-0acf-403d-a96a-f626ae2d6bf2) the fix?